types/p-log/index.d.ts

```diff
@@ -0,0 +1,13 @@
+// Type definitions for p-log 1.0
+// Project: https://github.com/sindresorhus/p-log#readme
+// Definitions by: BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = pLog;
+
+declare const pLog: pLog;
+
+interface pLog {
+    <T>(): (value: T) => Promise<T>;
```

Comment:
the location of this type parameter declaration looks wring to me. either, `(): (value: any) => Promise<any>;` or (): <T>(value: T) => Promise<T>;`